packages/modular-scripts/src/program.ts

```diff
@@ -155,6 +164,19 @@ program
   .allowUnknownOption()
   .description('Run tests over the codebase')
   .action(async (regexes: string[], options: CLITestOptions) => {
+    if (options.ancestors && !options.changed) {
+      process.stderr.write(
+        "Option --ancestors doesn't make sense without option --changed",
+      );
+      process.exit(1);
+    }
+    if (options.changed && regexes.length) {
+      process.stderr.write(
+        'Option --changed conflicts with supplied test regex',
+      );
+      process.exit(1);
+    }
+
     const { default: test } = await import('./test');
```

Comment:
Does importing `./test` have side-effects?

Reply:
nope

Reply:
In that case it doesn't need a dynamic import - not something that's part of your PR, but might be easier to follow as a standard import at the top of the file.

Reply:
@steveukx in `program.ts` we parse the command line args and we decide which Modular command to execute. If we statically `import`ed all of them at the start (which is translated to sync `require` in cjs world), it'd be a performance hit proportional to the quantity of commands.